### PR TITLE
Merge final `release/19.9`

### PR DESCRIPTION
**Pull request by ParaskP7**
Final release of `19.9`.

 - [x] #16648
 - [x] #16651
 - [x] `values*/strings.xml` updated with latest translations
 - [x] `metadata` and `jetpack_metadata` files updated with latest translations
 - [x] Version bumped in `version.properties`
 - [x] WPAndroid submitted to Google for review
 - [x] JPAndroid submitted to Google for review

### Conflicts

1. `build.gradle` (Gutenberg Mobile Version)

<DETAILS>

Gutenberg version conflict:
```
<<<<<<< release/19.9
    gutenbergMobileVersion = 'v1.76.2'
=======
    gutenbergMobileVersion = 'v1.77.0'
>>>>>>> trunk
```

And I kept `gutenbergMobileVersion = 'v1.77.0'`, which was in `trunk`. The `trunk`'s version seems to be the latest out of the two and thus the version we want to keep. @SiobhyB , can you please verify and confirm that the resolution is what you would have expected as well? 🙇 

</DETAILS>

2. `build.gradle` (FluxC Version)

<DETAILS>

FluxC version conflict:
```
<<<<<<< release/19.9
    fluxCVersion = '1.42.3'
=======
    fluxCVersion = 'trunk-f72e5d145dcfcf6e6875863e874496c1866676e7'
>>>>>>> trunk
```

And I kept `fluxCVersion = '1.42.3'`, which was in `release/19.9`. The `release/19.9`'s version seems to be the latest out of the two and thus the version we want to keep. @ashiagr , can you please verify and confirm that the resolution is what you would have expected as well? 🙇 

</DETAILS>

### CODE REVIEW

**Comment by SiobhyB** [comment]
> And I kept gutenbergMobileVersion = 'v1.77.0', which was in trunk. The trunk's version seems to be the latest out of the two and thus the version we want to keep. @SiobhyB , can you please verify and confirm that the resolution is what you would have expected as well? 🙇

@ParaskP7, that's right, that's the one we want to keep. :) Thanks for wrangling this!

**Comment by ParaskP7** [comment]
👋 @AliSoftware @ashiagr !

FYI: The build in not working because we probably need to create a new and more recent `FluxC` tag. This also relates to what @oguzkocer noted in this [tmp PR](https://github.com/wordpress-mobile/WordPress-Android/pull/16658) he opened yesterday about updating `FluxC` to `1.44.0` and the [merge PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2430) he did on `FluxC`'s trunk.

**Comment by AliSoftware** [comment]
```
<<<<<<< release/19.9
    fluxCVersion = '1.42.3'
=======
    fluxCVersion = 'trunk-f72e5d145dcfcf6e6875863e874496c1866676e7'
>>>>>>> trunk
```
> And I kept fluxCVersion = '1.42.3', which was in release/19.9. The release/19.9's version seems to be the latest out of the two and thus the version we want to keep. @ashiagr , can you please verify and confirm that the resolution is what you would have expected as well? 🙇

In theory I'd agree, because `1.42.3` is basically FluxC's `f72e5d1` (what WPAndroid's `trunk` points to) + the commit `adb9cb9` from wordpress-mobile/WordPress-FluxC-Android#2426. So making `trunk` point to `1.42.3` would just add `adb9cb9` on top of what `trunk` was using.

image

But it's also worth noting that the build of this PR fails on CI (I didn't check if that was because of FluxC API mismatching though? But I guess it is likely to be so).

And given that, once this PR lands in `trunk`, we'll do the code-freeze for WPAndroid 20.0, and we will then need to update `fluxCVersion` to `1.44.0` — the version that Oguz created when he did the WooCommerce code freeze (see also his notes in https://github.com/wordpress-mobile/WordPress-Android/pull/16658) — we might as well try to resolve the CI failure and merge conflict by pointing to `1.44.0` right away in this PR (and fix the build failures accordingly) rather than point to `1.42.3` in this PR and fix those failures, then point to `1.44.0` in the upcoming code freeze and have to fix other failures again.

Hoping that @ashiagr can help with that so that we land this ASAP to not delay the code freeze of 20.0 (which depends on this landing) too much longer 🙏 🤞

**Comment by ParaskP7** [comment]
Yes, thanks your input and elaborate response @AliSoftware ! 🙇 

I too believe that the best course of action is to point `FluxC` to `1.44.0` as you (and @oguzkocer) suggested, and doing that in this PR. However, I think it is best to double check that with @ashiagr before doing so. 💯

**Comment by ashiagr** [comment]
> best course of action is to point FluxC to 1.44.0 as you (and @oguzkocer) suggested, and doing that in this PR. However, I think it is best to double check that with @ashiagr before doing so. 

I can see changes from `1.42.3` included in `1.44.0` tag here: https://github.com/wordpress-mobile/WordPress-FluxC-Android/commit/89f84603265524c01b80f772f07ef6dbd4dc4a6d
I'm on board with the suggestion.

**Comment by AliSoftware** [comment]
> > best course of action is to point FluxC to 1.44.0 as you (and @oguzkocer) suggested, and doing that in this PR. However, I think it is best to double check that with @ashiagr before doing so.
> 
> I can see changes from `1.42.3` included in `1.44.0` tag here: [wordpress-mobile/WordPress-FluxC-Android@89f8460](https://github.com/wordpress-mobile/WordPress-FluxC-Android/commit/89f84603265524c01b80f772f07ef6dbd4dc4a6d) I'm on board with the suggestion.

@ashiagr the thing is that `1.44.0`, in addition to include changes from `1.42.3`, also include changes from #2423, #2413, #2424, #2328 — that were **not** yet integrated in WPAndroid's `trunk`. So pointing FluxC to `1.44.0` will also include those 4… while those were never used by neither `release/19.9` nor by `trunk`.

I still think the right move is to point to FluxC `1.44.0` here to resolve the conflict, and to avoid continuing the divergence of tags between WC and WP any further. But doing that will require your help to:
 - Fix any potential build failure in WPAndroid that would be related to including those 4 FluxC PRs that were not in WPAndroid yet
 - Ensuring that including those 4 FluxC PRs would not break anything in WPAndroid. e.g. if one of those 4 PRs changes an existing API that was previously used by WPAndroid — especially if that change in FluxC was done for a WIP feature of WPAndroid that has not landed in WPAndroid's trunk yet and is still in progress — that would mean changing the behavior of WPAndroid or including things that were still WIP and not yet ready to land in WPAndroid's trunk…

This is why we'll need your help to fix the build issues but also confirm or deny the impact of including those 4 additional FluxC PRs into trunk, as this is not as trivial/impactless as it might seem at first.

**Comment by ashiagr** [comment]
Thanks for the explanation @AliSoftware! I am just checking changes from https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2413 which might cause issues. Others look fine to me. I'll shortly provide an update.

**Comment by ParaskP7** [comment]
FYI: @ashiagr I tried building this release branch with `1.44.0` and it actually not compiling as `ACTION_GROW`, `ACTION_REMINDER` and `ACTION_SCHEDULE` are missing on `InsightsManagementMapper.toName(...)` and its `when` expression since it must to be exhaustive.

PS: At this point I am shamelessly adding @ravishanker to this discussion as I am seeing this [Refactor logic](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2413/commits/7d2a28b256423103a7f3f31dfdc6a4621ba5f7ff) commit, which is part of this [Stats Revamp v2: Add action types #2413](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2413/commits) FluxC PR that is causing that. Maybe it will help. 🤞

**Comment by ashiagr** [comment]
I see that `WPAndroid` PR https://github.com/wordpress-mobile/WordPress-Android/pull/16608 corresponding to `FluxC` changes in https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2413 is not merged to `WPAndroid` `trunk` yet, however FluxC PR is merged to its `trunk`.  Ideally, we should not merge a `FluxC` PR until the corresponding client app PR is ready to be merged. 

Does it help if 
- I create `FluxC` PR to revert https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2413 from `trunk`
and
- Once it is merged to `FluxC` `trunk`, point `trunk` hash in this PR?

**Comment by AliSoftware** [comment]
@ashiagr that seems like the best course of action moving forward indeed! Good idea 👍 

I'll let you do the revert on FluxC, point this PR to the new `trunk-<sha1>`, and ping us once it's done so we can merge.

PS: Once this PR is merged on WPAndroid's `trunk`, I'll do a new FluxC version `1.44.1` pointing to that same new `trunk-<sha1>` (because we want our `release/*` branches post-code-freeze to point to tags), and by then you'll be able to re-open yet another PR on FluxC to prepare the re-introduction of 2313 in FluxC… but wait for https://github.com/wordpress-mobile/WordPress-Android/pull/16608 to be merged in WPAndroid before merging it in FluxC

**Comment by ashiagr** [comment]
@ParaskP7, @AliSoftware 

I've updated `FluxC` version to new `trunk-<sha1>` in d6e6141. I've also tested locally that build issues are resolved. Hopefully, we'll be able to merge this PR once CI completes. 🙏 🤞

**Comment by ParaskP7** [comment]
Many many many thanks @ashiagr !!! 🙇 🙇 🙇

File: build.gradle
Comment (AliSoftware) [inline build.gradle:15]: For refs for people coming back to that PR in the future: this is the revert we talked about in the comments above, see [#2432](https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/2432)

**Review by AliSoftware — APPROVED** [review]

**Comment by AliSoftware** [comment]
> Many many many thanks @ashiagr !!! 🙇 🙇 🙇

Indeed! 🙇‍♂️

**Pull request merged**
